src/pip/_internal/resolution/resolvelib/requirements.py

```diff
@@ -33,8 +35,14 @@ def name(self):
         # No need to canonicalise - the candidate did this
         return self.candidate.name
 
-    def find_matches(self):
-        # type: () -> Sequence[Candidate]
+    def find_matches(self, is_root, constraints):
+        # type: (bool, Sequence[SpecifierSet]) -> Sequence[Candidate]
+        if constraints:
+            raise InstallationError(
+                "Could not satisfy constraints for '{}': "
+                "installation from path or url cannot be "
+                "constrained to a version".format(self.name)
+            )
```

Comment:
Is this to match the legacy resolver’s behaviour? Because to me it makes sense to allow this (if the version satisfies the constraints, of course).